tasks/main.yml

```diff
@@ -24,6 +24,48 @@
   when:
     - network_state is defined
     - ansible_distribution_major_version | int < 8
+
+- name: Check if updates for network packages are available through the DNF
+    package manager due to wireless or team interfaces
+  dnf:
+    update_cache: true
+    name: "{{ network_packages }}"
+    state: latest  # noqa package-latest
+  register: dnf_package_update_info
+  check_mode: true
+  when:
+    - ansible_distribution == 'Fedora' or
+      ansible_distribution_major_version | int > 7
+    - __network_wireless_connections_defined
+      or __network_team_connections_defined
+
+- name: Check if updates for network packages are available through the YUM
+    package manager due to wireless or team interfaces
+  yum:
+    update_cache: true
+    name: "{{ network_packages }}"  # noqa package-latest
+    state: latest
+  register: yum_package_update_info
+  check_mode: true
+  when:
+    - ansible_distribution_major_version | int < 8
+    - __network_wireless_connections_defined
+      or __network_team_connections_defined
+
+- name: Ask user's consent to restart NetworkManager due to wireless or team
+    interfaces
+  fail:
+    msg: NetworkManager needs to be restarted to be able to proceed. This might
+      disturb the connectivity of the managed system. Please set
+      `network_allow_restart` to `true` if you are prepared for this.
```

Comment:
Please explain to the user that this is because wireless or team interfaces are defined. Also, the full necessary action is to install the package and to restart NM. It seems best to be explicit about this, so users can also install the packages/restart themselves if necessary.